### PR TITLE
108: Add request body and headers as differentiators for requests

### PR DESCRIPTION
**Pull request by cialvarez**
## Objectives
- To allow VCRURLConnection to work with requests that have different bodies and headers

## How to Test
- Run unit tests; all should pass.

New unit tests added:
- `testThatRequestsWithDifferentBodiesAreNotEqual`
- `testThatRequestsWithDifferentHeadersAreNotEqual`
- `testThatRequestsWithSameBodiesAreEqual`
- `testThatRequestsWithSameHeadersAreEqual`

### CODE REVIEW

File: Tests/VCRURLConnectionTests/VCRRequestKeyTests.m [hunk 1]
```diff
@@ -48,6 +48,73 @@ - (void)testIsEqual {
...
+    VCRRequestKey *requestKeyFromRecording = [VCRRequestKey keyForObject:recording];
+
+    XCTAssertNotEqualObjects(requestKeyFromURLRequest, requestKeyFromRecording, @"Key objects should be equal");
+}
+
+- (void)testThatRequestsWithDifferentHeadersAreNotEqual {
+    NSString *path = @"http://foo/bar";
+    NSURL *url = [NSURL URLWithString:path];
+    NSMutableURLRequest *request = [[NSMutableURLRequest alloc] initWithURL:url];
+    request.allHTTPHeaderFields = @{ @"Omae wa mou shindeiru": @"nani" };
```
Comment (luisligunas) [inline Tests/VCRURLConnectionTests/VCRRequestKeyTests.m:71]: Nice.

File: Tests/VCRURLConnectionTests/VCRRequestKeyTests.m [hunk 2]
```diff
@@ -48,6 +48,73 @@ - (void)testIsEqual {
...
+    request.HTTPBody = [@"Omae wa mou shindeiru" dataUsingEncoding:NSUTF8StringEncoding];
+    VCRRequestKey *requestKeyFromURLRequest = [VCRRequestKey keyForObject:request];
+
+    VCRRecording *recording = [[VCRRecording alloc] init];
+    recording.method = @"GET";
+    recording.URI = path;
+    recording.requestData = [@"yamete kudasai" dataUsingEncoding:NSUTF8StringEncoding];
+    VCRRequestKey *requestKeyFromRecording = [VCRRequestKey keyForObject:recording];
+
+    XCTAssertNotEqualObjects(requestKeyFromURLRequest, requestKeyFromRecording, @"Key objects should be equal");
```
Comment (luisligunas) [inline Tests/VCRURLConnectionTests/VCRRequestKeyTests.m]: I think it should "Key objects should not be equal".

File: Tests/VCRURLConnectionTests/VCRRequestKeyTests.m [hunk 3]
```diff
@@ -48,6 +48,73 @@ - (void)testIsEqual {
...
+    request.allHTTPHeaderFields = @{ @"Omae wa mou shindeiru": @"nani" };
+    VCRRequestKey *requestKeyFromURLRequest = [VCRRequestKey keyForObject:request];
+
+    VCRRecording *recording = [[VCRRecording alloc] init];
+    recording.method = @"GET";
+    recording.URI = path;
+    recording.requestHeaderFields = @{ @"Yamete": @"Kudasai" };
+    VCRRequestKey *requestKeyFromRecording = [VCRRequestKey keyForObject:recording];
+
+    XCTAssertNotEqualObjects(requestKeyFromURLRequest, requestKeyFromRecording, @"Key objects should be equal");
```
Comment (luisligunas) [inline Tests/VCRURLConnectionTests/VCRRequestKeyTests.m]: I think it should "Key objects should not be equal".

**Review by luisligunas — APPROVED** [review]
Great work, Christian! LGTM (at least as far as I can understand 😂)!

File: Tests/VCRURLConnectionTests/VCRRequestKeyTests.m (hunk 2 above)
Comment (cialvarez) [reply Tests/VCRURLConnectionTests/VCRRequestKeyTests.m]: Good catch, @luisligunas !

**Pull request merged**